tests/robot/suites/crud/app_namespaces_crud.robot

```diff
@@ -0,0 +1,211 @@
+    Add Agent VPP Node    agent_vpp_1
+    Sleep    ${RESYNC_WAIT}
+
+Get Interfaces Sw If Index After Resync 1
+    ${TAP1_SW_IF_INDEX}=    vpp_ctl: Get Interface Sw If Index    agent_vpp_1    ${TAP1_NAME}
+    ${TAP1_SW_IF_INDEX}=    Convert To String    ${TAP1_SW_IF_INDEX}
+    Set Suite Variable    ${TAP1_SW_IF_INDEX}
+    ${TAP2_SW_IF_INDEX}=    vpp_ctl: Get Interface Sw If Index    agent_vpp_1    ${TAP2_NAME}
+    ${TAP2_SW_IF_INDEX}=    Convert To String    ${TAP2_SW_IF_INDEX}
+    Set Suite Variable    ${TAP2_SW_IF_INDEX}
+
+Check NS1, NS2 And NS3 Were Automatically Configured After Resync
+    vpp_term: Check Data In Show Application Namespaces Output    agent_vpp_1    ${NS1_ID}    1    ${SECRET2}    ${TAP1_SW_IF_INDEX}
+    vpp_term: Check Data In Show Application Namespaces Output    agent_vpp_1    ${NS2_ID}    2    ${SECRET1}    ${TAP2_SW_IF_INDEX}
+    vpp_term: Check Data In Show Application Namespaces Output    agent_vpp_1    ${NS3_ID}    3    ${SECRET4}    ${TAP2_SW_IF_INDEX}
+
+Disable L4 Features And Check They Are Disabled
+    vpp_ctl: Set L4 Features On Node    node=agent_vpp_1    enabled=false
+    ${out}=    vpp_term: Show Application Namespaces    node=agent_vpp_1
+    Should Contain    ${out}    show app: session layer is not enabled
+
+Put Namespace NS4 While L4 Features Are Disabled
+    vpp_ctl: Put Application Namespace    node=agent_vpp_1    id=${NS4_ID}    secret=${SECRET4}    interface=${TAP1_NAME}
+
+Enable L4 Features And Check Namespaces NS1, NS2, NS3 And NS4 Are Present In Namespaces List
+    vpp_ctl: Set L4 Features On Node    node=agent_vpp_1    enabled=true
+    ${out}=    vpp_term: Show Application Namespaces    node=agent_vpp_1
+    vpp_term: Check Data In Show Application Namespaces Output    agent_vpp_1    ${NS1_ID}    1    ${SECRET2}    ${TAP1_SW_IF_INDEX}
+    vpp_term: Check Data In Show Application Namespaces Output    agent_vpp_1    ${NS2_ID}    2    ${SECRET1}    ${TAP2_SW_IF_INDEX}
+    vpp_term: Check Data In Show Application Namespaces Output    agent_vpp_1    ${NS3_ID}    3    ${SECRET4}    ${TAP2_SW_IF_INDEX}
+    vpp_term: Check Data In Show Application Namespaces Output    agent_vpp_1    ${NS4_ID}    4    ${SECRET4}    ${TAP1_SW_IF_INDEX}
+
+Put Namespace NS5 Associated With MEMIF1 Interface That Is Not Created And Check NS5 Is Not Present In Namespaces List
+    vpp_ctl: Put Application Namespace    node=agent_vpp_1    id=${NS5_ID}    secret=${SECRET1}    interface=${MEMIF1_NAME}
+    ${out}=    vpp_term: Show Application Namespaces    node=agent_vpp_1
+    Should Not Contain    ${out}    ${NS5_ID}
+
+Put MEMIF1 Interface And Check Namespace NS5 Is Present In Namespaces List
+    vpp_ctl: Put Memif Interface With IP    node=agent_vpp_1    name=memif1    mac=${MEMIF1_MAC}    master=true    id=1    ip=192.168.1.2    prefix=28    socket=default.sock
+    vpp_term: Check Data In Show Application Namespaces Output    agent_vpp_1    ${NS5_ID}    5    ${SECRET1}    ${MEMIF1_SW_IF_INDEX}
```

Comment:
might be timing  problem in some cases,  use sleep at least 1s for the configuration ( in general) to take effect before the check